installer/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/SaaSBoostInstall.java

```diff
@@ -959,15 +1028,72 @@ protected void deleteApplicationConfig() {
-        // Also, this won't scale super well and could be parallelized...
-        String tenantStackName = "Tenant-" + ((String) tenant.get("id")).split("-")[0];
-        outputMessage("Deleteing tenant CloudFormation stack " + tenantStackName);
-        deleteCloudFormationStack(tenantStackName);
+        // TODO we can parallelize to improve performance with lots of tenants
+        String tenantStackId = (String) ((HashMap<String, Object>)((HashMap<String, Object>) tenant.get("resources")).get("CLOUDFORMATION")).get("arn");
+        final ObjectMapper mapper = new ObjectMapper();
+        try {
+            Map<String, Object> systemApiRequest = new HashMap<>();
+            Map<String, Object> detail = new HashMap<>();
+            detail.put("resource", "tenants/" + (String) tenant.get("id"));
+            detail.put("method", "DELETE");
+            Map<String, String> tenantIdOnly = new HashMap<>();
+            tenantIdOnly.put("id", (String) tenant.get("id"));
+            detail.put("body", mapper.writeValueAsString(tenantIdOnly));
+            systemApiRequest.put("detail", detail);
+            final byte[] payload = mapper.writeValueAsBytes(systemApiRequest);
+            try {
+                LOGGER.info("Invoking delete tenant API");
+                InvokeResponse response = lambda.invoke(request -> request
+                        .functionName("sb-" + this.envName + "-private-api-client")
+                        .invocationType(InvocationType.REQUEST_RESPONSE)
+                        .payload(SdkBytes.fromByteArray(payload))
+                );
+                if (response.sdkHttpResponse().isSuccessful()) {
+                    LOGGER.error("got response back: {}", response);
+                    // wait for tenant CloudFormation stack to reach deleted
+                    boolean waiting = true;
+                    while (waiting) {
+                        DescribeStacksResponse stackStatusResponse = cfn.describeStacks(request -> request.stackName(tenantStackId));
+                        StackStatus stackStatus = stackStatusResponse.stacks().get(0).stackStatus();
+                        switch (stackStatus) {
+                            case DELETE_COMPLETE:
+                            case DELETE_FAILED: {
+                                waiting = false;
+                                break;
+                            }
+                            case DELETE_IN_PROGRESS: {
+                                outputMessage("Waiting 1 minute for " + tenantStackId + " to finish deleting.");
+                                try {
+                                    Thread.sleep(60 * 1000);
+                                } catch (InterruptedException e) {
+
+                                }
+                            }
+                            default: {
+                                outputMessage("Unexpected stackStatus " + stackStatus + " while waiting for " + tenantStackId + " to finish deleting.");
```

Comment:
Not exactly unexpected since the application services stacks must finish deleting before the base stack for the tenant will start deleting.
We should probably have some sort of maximum wait timeout.